### ISSUE TITLE
Typegen is generating erroneous code

### ISSUE THREAD

**Issue by niklabh** [polkadot-js/api#5503]
* **I'm submitting a ...**

  - [x] Bug report
  - [ ] Feature request
  - [ ] Support request
  - [ ] Other

* **What is the current behavior and expected behavior?**

We are using @polkadot/typegen 9.13.6 for TypeScript auto generated code for Chain modules: [script](https://github.com/AstarNetwork/astar.js/blob/main/packages/types/scripts/generate.ts). We are getting errors as code generated includes many new types taken from chain but they are not updated and imported.

![image](https://user-images.githubusercontent.com/874046/220396721-fa584057-b0de-4ca8-8f6a-7fe21deb886b.png)

* **Please tell us about your environment:**

  - Version:
  - Environment:

    - [x] Node.js v18.12.0
    - [ ] Browser
    - [ ] Other (limited support for other environments)

  - Language:

    - [ ] JavaScript
    - [x] TypeScript (include tsc --version) 4.6.2
    - [ ] Other

**Comment by jacogr** [comment]
The latest released version actually had a fix for storage that popped-up on Polkadot. It could be related.

**Comment by jacogr** [comment]
Ok, so on "new types, not reflected", this is actually known and I stumbled across it this week.

The issue is that the generator processes are quite seperate and they would directly import the types. so basically what happens.

1. Lookup type generation runs, exporting the new types
2. Augmentation generation runs, still using the types from before lookup

In the case of this repo, it runs into that around once every 6 weeks - it follows Substrate/Polkadot/Kusama with weekly updates. To get around it, it means run and then re-run to pickup the now-available-since-previous-generation runs.

As it stands, I'm not going to touch this - it is a slight PITA, but it is what it is. Happy with PRs to re-order stuff (that still works), but since it is a not-everytime case, with a work-around (re-run), it is not something that has actually become painful enough for me to spend time and try and dissect re-order (the ordering is challenging anyway).

If somebody ever wants to spend time, I believe the crux of the matter is that when running augmentation, it needs to re-read on-disk (assuming the lookup/interface generation has taken place - that _could_ also be out of order)

So as it stands - closing, but would certainly look at PRs making attempts to make this better.

TL;DR Workaround available, not pretty, but also not painful enough to keep in the queue.
